### PR TITLE
Document naming conventions & Naming alignment

### PR DESCRIPTION
**Pull request by jfaltermeier**
Please check the `Naming.md` first https://github.com/eclipse-theia/theia-cloud/pull/368/files#diff-458dcaeb7b373b080fb85240d56af356cc0cbc6d054f90fb92ef87dcf4748cb5

Let me know what you think about the changes. 

Contributed on behalf of STMicroelectronics

Helm Chart Changes

https://github.com/eclipse-theia/theia-cloud-helm/pull/78

### CODE REVIEW

File: documentation/Naming.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+- **NPM Package names**
+- **File names**
+
+## `THEIA_CLOUD`
+
+Use **all uppercase** with underscores for **environment variables**, such as `THEIA_CLOUD_SESSION_NAME`.
+
+## Theia Cloud
+
+In general documentation and free text, use the standard project name, **Theia Cloud** (with a space). Avoid spaces in technical contexts.
```
Comment (lucas-koehler) [inline documentation/Naming.md:37]:
```suggestion
In general documentation and free text, use the standard project name **Theia Cloud** (with a space). Avoid spaces in technical contexts.
```

File: documentation/Naming.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Use **PascalCase** or **camelCase** for Java/TypeScript class, method, and property names, such as `DefaultTheiaCloudOperatorLauncher` or `window.theiaCloudConfig`.
+
+This format is also appropriate in contexts where spaces are avoided for technical reasons, such as in **Keycloak Realm** names (e.g., `TheiaCloud`).
+
+## `theia-cloud`
+
+This format serves as a preferred technical abbreviation. Use it when no other conventions apply, including:
+
+- **NPM Package names**
+- **File names**
```
Comment (sgraband) [inline documentation/Naming.md:29]:
```suggestion
- **File names**
- **k8s namespaces**
```

File: java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public class EagerStartAppDefinitionAddedHandler implements AppDefinitionHandler
 
     private static final Logger LOGGER = LogManager.getLogger(EagerStartAppDefinitionAddedHandler.class);
 
-    public static final String LABEL_KEY = "theiacloud";
+    public static final String LABEL_KEY = "theia-cloud.io/template-purpose";
```
Comment (lucas-koehler) [inline java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java:60]: We should be consistent with word separators also after the prefix.
Here we use a dash in `template-purpose`.
However, in other places we use a label like `theia-cloud.io/workspace.name: placeholder-label-workspace-name`.

We should use either dot or dash consistently. I slightly prefer dashes as they seem more typical in a uri resp. dns name format to me. WDYT?

**Review by lucas-koehler — CHANGES_REQUESTED** [review]
Hi @jfaltermeier, thanks for aligning the naming and the documentation. This saves a lot of wondering what is the correct spelling :sparkles: 

The changes already look pretty good to me. I have two remarks inline. One tiny suggestion and one hint regarding consistency in Kubernetes label key naming. Please have a look.

File: node/monitor/package.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "theiacloud-monitor",
+  "name": "theia-cloud-monitor",
```
[the remark this answers is not in the document]
Comment (jfaltermeier) [reply node/monitor/package.json:2]: The npm package is already published under a name following the conventions: https://www.npmjs.com/package/@eclipse-theiacloud/monitor-theia. 
This one here is the vsix, which hasn’t been published to open-vsx yet, so we should be fine.

File: java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java (hunk 3 above)
Comment (jfaltermeier) [reply java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java:60]: Sounds good, I will adjust this

**Comment by jfaltermeier** [comment]
We have to use `THEIACLOUD` for env variables, because Theia messes with variables starting with `THEIA_`

I addressed this + the comments above now

File: java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java (hunk 3 above)
Comment (jfaltermeier) [reply java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/handler/appdef/EagerStartAppDefinitionAddedHandler.java:60]: I updated `theia-cloud.io/workspace.name` to `theia-cloud.io/workspace-name`

**Comment by jfaltermeier** [comment]
This is the theia code: https://github.com/eclipse-theia/theia/blob/f29c3d04241b32791462c7027938abeb887f8140/packages/core/src/node/messaging/ipc-protocol.ts#L67

**Review by lucas-koehler — APPROVED** [review]
Thanks for the updates. LGTM now :sparkles:

**Pull request merged**